File: php/bubble_sort.php
<?php
function bubbleSort(&$arr) {
    $n = count($arr);
    for ($i = 0; $i < $n - 1; $i++) {
        for ($j = 0; $j < $n - $i - 1; $j++) {
            if ($arr[$j] > $arr[$j + 1]) {
                $temp = $arr[$j];
                $arr[$j] = $arr[$j + 1];
                $arr[$j + 1] = $temp;
            }
        }
    }
}

echo "Enter the number of elements: ";
$n = intval(trim(fgets(STDIN)));

echo "Enter the elements (space separated): ";
$line = trim(fgets(STDIN));
$arr = array_map('intval', explode(' ', $line));

bubbleSort($arr);

echo "Sorted array: ";
foreach ($arr as $num) {
    echo $num . " ";
}
echo PHP_EOL;
?>

File: Java/BubbleSort.java
import java.util.Scanner;

public class BubbleSort {
    
    // time complexity: O(n^2)
    // space complexity: O(1)
    private static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // System.out.println("Enter the number of elements: ");
        int n = sc.nextInt();
        // System.out.println("Enter the elements: ");
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        bubbleSort(arr);

        // System.out.println("Sorted array: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();

        sc.close();
    }

    
}
